content/en/blog/_posts/2021-03-09-Defining-NetworkPolicy-Conformance-For-CNIs

```diff
@@ -0,0 +1,384 @@
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
++--------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+| z/a    | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
++--------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+| z/b    | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
++--------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+| z/c    | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
+|        | X   | .   | .   | .   | .   | .   | .   | .   | .   |
++--------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+Both Cyclonus and the e2e tests use the same strategy to validate a Network Policy of probing pods over TCP or UDP.
+As an example of how we use Cyclonus to help make CNI implementations better from a Network Policy perspective, you can see the following issues:
+
+- [Antrea: NetworkPolicy: unable to allow ingress by CIDR](https://github.com/vmware-tanzu/antrea/issues/1764)
+- [Calico: default missing protocol to TCP; don't let single port overwrite all ports](https://github.com/projectcalico/libcalico-go/pull/1373)
+- [Cilium: Egress Network Policy allows traffic that should be denied ](https://github.com/cilium/cilium/issues/14678)
+
+The good news is, all of these issues are actively being fixed and iterated on between SIG Network, the CNI providers, and the Network Policy subproject.
+
+If you're interested in verifying NetworkPolicy functionality on your cluster (and, if you care about security, or run a SaaS, you should be), then you can run the upstream end to end tests, or Cyclonus, or both.
+- If you are deeply curious about your CNI providers Network Policy implementation, use Cyclonus to test *100s* of policies, and evaluate your CNI for comprehensive functionality, for deep discovery of potential security holes, or for getting involved with the upstream networkpolicy efforts.
+- If your just getting started with Network Policies, your better off running the e2e tests if you want to simply verify the "common" Network Policy cases that most CNIs should be implementing correctly, in a way that is quick to diagnose.
+
+## Where to start with NetworkPolicy testing ?
+
+- You can run Cyclonus on your cluster easily by following the [instructions at](https://github.com/mattfenwick/cyclonus), and determine wether *your* specific CNI configuration is
+fully conformant to the 100s of different Kubernetes Network Policy API constructs.
```

Comment:
How about:
```suggestion
- You can run Cyclonus on your cluster easily by following the
  [instructions](https://github.com/mattfenwick/cyclonus#readme) in its
  README, and determine whether *your* specific CNI plugin configuration
  fully conforms to the hundreds of different Kubernetes NetworkPolicy
  API constructs that Cyclonus tests.
```
(if rejecting the suggestion, note that “wether” in the original is misspelled.

Reply:
👍